### ISSUE TITLE
Box selection [Mac]

### ISSUE THREAD

**Issue by kivicode** [epezent/implot#171]
Hello! I'm a bit confused with the box selection process. Firstly, there is a line that `ImPlotFlags_Selection` flag was changed to `ImPlotFlags_BoxSelect` (see `implot.cpp`), but I can't find neither statement that `ImPlotFlags_BoxSelect` has been deleted, nor that it's being used in the code. So the main question is how to enable box selection/zooming?

**Comment by epezent** [comment]
@kivicode , all flags including `ImPlotFlags_BoxSelect` were eventually changed as follows:

```
- 2020/09/06 (0.7) - Several flags under ImPlotFlags and ImPlotAxisFlags were inverted (e.g. ImPlotFlags_Legend -> ImPlotFlags_NoLegend) so that the default flagset
                     is simply 0. This more closely matches ImGui's style and makes it easier to enable non-default but commonly used flags (e.g. ImPlotAxisFlags_Time).
```

Thus the flag in question is now `ImPlotFlags_NoBoxSelect`. Therefore, unless you specifically enable this flag, box selection is enabled by default. Simply right click drag inside of the plot area :)

**Comment by epezent** [comment]
See also: https://github.com/epezent/implot/issues/48#issuecomment-687888961

**Comment by kivicode** [comment]
Thank you! I've just checked but `ImPlotFlags_NoBoxSelect` is false, but I can't use the box selection. May it be an OSX-related event handling problem or something like that?

**Comment by epezent** [comment]
Hmm, can you post the code you use to create the plot? It might be related to a change I made yesterday #170 but I doubt it. I don't currently have a mac to check on.

**Comment by kivicode** [comment]
Sure, I've made a [blank project](https://gist.github.com/kivicode/24a587a81fb0a5fd7bcf8de170dee62a) to test it. I've just realized that it actually works, but there is no border or other visual notification. So, now there is another question how to turn it on  😅

**Comment by epezent** [comment]
So the plot resizes when you right-click drag and release, but you cannot see the box drawn like so:

![bs](https://user-images.githubusercontent.com/29577475/105097703-abca5b00-5a6e-11eb-98c4-08e900c421f9.gif)

The only thing that comes to mind is that you've set the style variable `ImPlotCol_Selection` to a fully transparent color. Otherwise, it would seem to be a rendering issue with your backend. If you could, post a GIF of what is happening.

**Comment by epezent** [comment]
I'm wondering if it is related to the selection box have alpha enabled. It seems like there have been issues with the OpenGL2 backend in the past https://github.com/ocornut/imgui/issues/3000, which you are using.

**Comment by kivicode** [comment]
Hmm, alpha bending seems to be ok everywhere else. The gif was recorded with the latest backend from the original ImGui repo and the code snippet I've posted earlier ![implot-2](https://user-images.githubusercontent.com/31818467/105099160-44d39300-5abc-11eb-90cc-284e9b9059ad.gif)

**Comment by epezent** [comment]
That's bizarre. You might try removing the 25% alpha multiplier here and reporting back if this at least makes the box visible. Obviously this is not ideal, but it gets us closer to solving the issue:

https://github.com/epezent/implot/blob/c73509d6d13d72e4a1cb19f242e709a5df9f64cb/implot.cpp#L2282

**Comment by kivicode** [comment]
Unfortunately not, it didn't help

**Comment by epezent** [comment]
Ok thanks, then it is probably not a rendering issue. Possibly the code that renders the box is not getting reached for some reason.  I am not able to reproduce this issue here, so you'll need to step through this with a debugger or just throw some print statements in there. In particular I'd like to know if any of the `DrawList.AddRect...` calls are being made. 

https://github.com/epezent/implot/blob/c73509d6d13d72e4a1cb19f242e709a5df9f64cb/implot.cpp#L2275-L2301

**Comment by kivicode** [comment]
Okay, it's late at night here, so I'll go through it a bit later.

**Comment by epezent** [comment]
Sounds good. Thanks for your help!

**Comment by epezent** [comment]
I just tested this on a Raspberry Pi 4 (closest thing I have to a Unix type environment) using OpenGL2 and everything is still fine on my end. So confused by this. 

Can you try changing the box color by setting or pushing a new value for `ImPlotCol_Selection` and seeing if anything changes?

**Comment by kivicode** [comment]
I've finally found the problem! It turns out to be about ImGui default OSX backed realization rather then ImPlot itself. The problem was that during right mouse button dragging the IO mouse position wasn't updating (so the rect had zero area). I've solved it by simply adding `|| event.type == NSEventTypeRightMouseDragged` [here](https://github.com/ocornut/imgui/blob/ae0d2dd614903257820a365ca44881aa45e21e5d/backends/imgui_impl_osx.mm#L229). Now it works as it supposed to and I'm closing the issue.

Thank you!
